src/wp-includes/class-wp-scripts.php

```diff
@@ -281,6 +281,24 @@ public function do_item( $handle, $group = false ) {
 			$intended_strategy = '';
 		}
 
+		// If the script is in the header group, current output is the header, the intended strategy is `defer`, the
+		// actual strategy is not defer and all dependent scripts are in the footer - move this script to the footer.
+		if ( 'defer' === $intended_strategy && 'defer' !== $strategy && 0 === $this->groups[ $handle ] && 0 === $group ) {
+			// Check that all depedencies are in the footer.
+			$all_deps_in_footer = true;
+			foreach ( $this->get_dependents( $handle ) as $dep ) {
+				if ( isset( $this->groups[ $dep ] ) && 1 !== $this->groups[ $dep ] ) {
```

Comment:
For consistency with the above comparison with the group:

```suggestion
				if ( isset( $this->groups[ $dep ] ) && $this->groups[ $dep ] > 0 ) {
```